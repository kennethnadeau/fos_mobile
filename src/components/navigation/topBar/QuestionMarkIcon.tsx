import React, {FC} from 'react';
import QuestionMarkSvg from '@fos/assets/svg/questionMark.svg';
import {s} from 'react-native-size-matters';
import {TouchableOpacity} from 'react-native';

type Props = {
  onPress: () => void;
};

export const QuestionMarkIcon: FC<Props> = ({onPress}) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <QuestionMarkSvg
        // FIXME Add i18n key for Support
        accessibilityLabel="Support"
        accessible
        fill="white"
        height={s(30)}
        width={s(30)}
      />
    </TouchableOpacity>
  );
};
