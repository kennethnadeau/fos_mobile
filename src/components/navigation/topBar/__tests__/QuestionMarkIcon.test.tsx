import React from 'react';
import {render, RenderAPI} from '@fos/utils/testing';
import {QuestionMarkIcon} from '../QuestionMarkIcon';

describe('<QuestionMarkIcon />', () => {
  let questionMarkIcon: RenderAPI;

  beforeEach(() => {
    questionMarkIcon = render(<QuestionMarkIcon onPress={jest.fn()} />);
  });

  it('should show question mark icon', () => {
    const actual = questionMarkIcon.queryAllByA11yLabel('Support');

    expect(actual).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(questionMarkIcon).toMatchSnapshot();
  });
});
