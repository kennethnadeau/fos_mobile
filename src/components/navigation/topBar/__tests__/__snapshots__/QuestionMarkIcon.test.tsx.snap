// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionMarkIcon /> should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <SvgMock
      accessibilityLabel="Support"
      accessible={true}
      fill="white"
      height={64.28571428571428}
      width={64.28571428571428}
    />
  </View>
</RNCSafeAreaProvider>
`;
